src/zenml/zen_server/routers/server_endpoints.py

```diff
@@ -71,6 +73,51 @@ def server_info() -> ServerModel:
     return zen_store().get_store_info()
 
 
+@router.get(
+    LOAD_INFO,
+    response_model=ServerLoadInfo,
+)
+@handle_exceptions
+def server_load_info(_: AuthContext = Security(authorize)) -> ServerLoadInfo:
+    """Get information about the server load.
+
+    Returns:
+        Information about the server load.
+    """
+    import threading
+
+    # Get the current number of threads
+    num_threads = len(threading.enumerate())
+
+    store = zen_store()
+
+    if store.config.driver == "sqlite":
+        # SQLite doesn't have a connection pool
+        return ServerLoadInfo(
+            threads=num_threads,
+            db_connections_total=0,
+            db_connections_active=0,
+            db_connections_overflow=0,
+        )
+
+    from sqlalchemy.pool import QueuePool
+
+    # Get the number of active connections
+    pool = store.engine.pool
+    assert isinstance(pool, QueuePool)
+    idle_conn = pool.checkedin()
```

Comment:
I'm still not really sure whether this is actually computing what we think it is:

I'm going to paste the QueuePool Implementation here again:

```python
def checkedin(self) -> int:
        return self._pool.qsize()

    def overflow(self) -> int:
        return self._overflow if self._pool.maxsize else 0

    def checkedout(self) -> int:
        return self._pool.maxsize - self._pool.qsize() + self._overflow
```

The total connections right now are computed as follows:
```
con_in_pool + (pool_max_size - con_in_pool + overflow) + max(0, overflow)
= pool_max_size + overflow + max(0, overflow)
```

Is that really correct though?

`overflow` starts at `-pool_max_size` and gets incremented each time a connection is opened, and decreased each time a connection is closed (at least that's my understanding of it).

I think we need to consider two cases:
Case 1: The amount of connections are less or equal connections than `max_pool_size` -> `overflow <=0`
In this case, I think the computation for the total connections is fine.

Case 2: There are more connections than the pool size:
In this case, the total connections are `pool_max_size + overflow + max(0, overflow)` which just adds the overflow twice.

Reply:
that might explain why I've seen bigger connection numbers when overflow is reached

Reply:
Maybe `total` is just `checkedin() + checkedout()`?

Reply:
yes, quite right